### PR TITLE
Colipali - ready for review.

### PR DESCRIPTION
**Pull request by michaelfeil**

### CODE REVIEW

**Review by michaelfeil — COMMENTED** [review]
Ready for review. If you come across this PR, feel free to review.

File: libs/infinity_emb/infinity_emb/transformer/vision/torch_vision.py [hunk 1]
```diff
@@ -18,49 +27,106 @@
...
     import torch
 if CHECK_TRANSFORMERS.is_available:
-    from transformers import AutoModel, AutoProcessor  # type: ignore
+    from transformers import AutoConfig, AutoModel, AutoProcessor  # type: ignore
+if CHECK_PIL.is_available:
+    from PIL import Image
 
 
-class ClipLikeModel(BaseClipVisionModel):
+class TorchImageModel(BaseClipVisionModel):
```
Comment (wirthual) [inline libs/infinity_emb/infinity_emb/transformer/vision/torch_vision.py:35]: For consistency, wouldn't make sense to also rename the base class here?

Comment (michaelfeil) [reply libs/infinity_emb/infinity_emb/transformer/vision/torch_vision.py]: Resolved!

**Comment by sir3mat** [comment]
Are there any tutorials on how to expose Colpali as a microservice?

**Comment by michaelfeil** [comment]
No, but you can deploy it via docker.

i would suggest you to get started with 

https://huggingface.co/michaelfeil/colpali-v12-random-testing (small one for testing the Api -enpoint)
https://huggingface.co/michaelfeil/colqwen2-v0.1 (for actual usage)

The following command after pip install and via docker works.

infinity_emb v2 —model-id michaelfeil/colpali-v12-random-testing

**Pull request merged**